### DISCUSSION TITLE
Load bodies outside xml file

### DISCUSSION THREAD

**Discussion by rohit-kumar-j** [google-deepmind/mujoco#321] [Asking for Help] [answered]
Can we load bodies into the simulator outside what is mentioned within the XML file?
Example:
```python
model = mujoco.MjModel.from_xml_path("path_to_xml")

bodies_index = []
for i in range(0,10):
    body_index = mujoco.loadURDF("path_to_stl_or_xml") # returns the unique body index
    bodies_index.append(body_index)
```
similar to [`pybullet.loadURDF()`](https://docs.google.com/document/d/10sXEhzFRSnvFcl3XxNGhnD4N2SedqwdAvK3dsihxVUA/edit#)

**Comment by yuvaltassa** [comment, marked as answer]
Hi,
I think you're asking for procedural modification of MuJoCo models? This is currently best supported by `dm_control`'s [pyMJCF](https://github.com/deepmind/dm_control/blob/main/dm_control/mjcf/README.md) module. If you have any questions, please follow up in the `dm_control` repo.
